src/CompilerWarnings.cmake

```diff
@@ -69,8 +69,7 @@ function(
     )
   endif()
 
-  if(WARNINGS_AS_ERRORS STREQUAL TRUE)
-    message(AUTHOR_WARNING "NOTE: WARNINGS_AS_ERRORS=${WARNINGS_AS_ERRORS}")
+  if(WARNINGS_AS_ERRORS STREQUAL "TRUE")
```

Comment:
Is “TRUE” the only acceptable positive value here? What if the user sets WARNINGS_AS_ERRORS to “ON” or “1”?

According to https://cmake.org/cmake/help/latest/command/if.html#basic-expressions, you can use `if(WARNINGS_AS_ERRORS)` to check for any of the ‘true’ constants, which are “ 1, ON, YES, TRUE, Y, or a non-zero number”. I’m not sure if you need the `${}`.

Reply:
Good point, it got caught up in the fixes I was making to fix the warning overrides.

Reply:
Good point, it got caught up in the fixes I was making to fix the warning overrides.

Reply:
584f03f